src/features/device/deviceSlice.ts

```diff
@@ -8,10 +8,9 @@ export interface IDeviceState {
   activeNode: MeshNode["data"]["num"] | null;
   availableSerialPorts: string[] | null;
   activeSerialPort: string | null;
-  activeWaypoint: Waypoint["id"] | null;
-  waypointEdit: boolean; // Controls if the waypoint edit menu, or the normal menu shows up on map
+  activeWaypoint: Waypoint | null;
```

Comment:
So the way that the WaypointEdit pane is set up, it needs a complete waypoint object. The WaypointEdit pane shows the info only for the active waypoint. I made `activeWaypoint` be Waypoints so that all of the waypoints that WaypointEdit potentially access are accessed in the same way.


The other approach I tried was to keep `activeWaypoint` as a `Waypoint['id']`, and also have a `tempWaypoint` state in redux that either stores a single complete temporary waypoint or `null`. Then for WaypointEdit we'd do a check for if the activeWaypoint ID was 0 to see if we needed to access the `tempWaypoint` state. 

I decided against this because I want to eventually combine `activeNode` and `activeWaypoint` - since there can only be one active whatever at a time, and we want to minimize the possibility of forgetting to update one when changing the other in the future. If both are just identified as ID, there could be a case where the IDs are the same for a radio and a waypoint.

Would love to hear thoughts though

Reply:
Couldn't we write the `selectActiveWaypoint` selector similarly to the `selectActiveNode` selector? I'm not positive I'm completely understanding the problem.

On the combining of `activeNode` and `activeWaypoint`, that's an interesting idea! I haven't really thought about that, but I'm wondering if we could set some field up where the field would both tell what pane was active as well as all the information needed to render that pane. Let's discuss that at some point!

Reply:
Hmm okay so the core issue is the new waypoint needing to be stored before being sent? That flow sounds correct for selecting the active node, although it doesn't "pull" from the radio as much as it just recalls the last data it heard from the radio. I think the clarification I would make is that we don't rely on the device to store any waypoint or text data for us, we just record all of the messages that we hear from the device and store them in the Rust backend.

Now that I'm thinking about it, would the temporary waypoint even be something we would want/need in Redux? My understanding is that the new waypoint is created and transmitted from within the `WaypointEditMenu` component, in the event that we're creating a new waypoint could we just hold the waypoint within the local component state? Once it gets sent by the device, the waypoint would end up in the redux state since we echo all waypoint and text packets back to the client.

Reply:
So the new waypoint is currently being created from the `MapView` component, then passed to redux. Then `WaypointEditMenu` finds the active waypoint from redux (similar to how the `MapSelectedNodeMenu` works). 

I'd push back against storing the info in a local component state - there are two components that depend on a complete temporary waypoint from `MapView`: the map pin (`Waypoint.tsx`) and the menu (`WaypointEditMenu`). If we cancel creating the waypoint from `WaypointEditMenu,` or add in emoji handling later, we'd also have to update `Waypoint.tsx`, rather than just having both get info from redux - I think it's messier than just having another state

Reply:
Time for me to (in barkin's words) cry over git again 

_(I know what to do, that was a joke)_